pkg/detectors/snowflake/snowflake_test.go

```
@@ -5,15 +5,20 @@ import (
 	"fmt"
 	"testing"
 
-	"github.com/brianvoe/gofakeit/v7"
 	"github.com/google/go-cmp/cmp"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/engine/ahocorasick"
 )
 
 func TestSnowflake_Pattern(t *testing.T) {
-	username := gofakeit.Username()
-	password := gofakeit.Password(true, true, true, false, false, 10)
```

IMO, using a hard-coded value is a step backwards. The generated passwords failing likely indicate a pattern issue, or perhaps a test setup issue.

---

Thanks for pointing this out. I was looking into the code of gofakeit module. The boolean as inputs control what characters are allowed in generated password but not guaranteed the inclusion. The snowflake password has a logic to ignore password which does not meet [the criteria](https://github.com/trufflesecurity/trufflehog/blob/b2ed7caeec8c1735fd5dd810095c531d877ac987/pkg/detectors/snowflake/snowflake.go#L88). When the generated password does not meet snowflake standard, the test gets break. 

For the same of simplicity, Like other pattern tests, I lean toward using hard-coded values. we could have write custom password generator to make sure it meets all the criteria. I avoided the complex path. 

Let me know what are your thoughts on this.

---

Hard-coded values are a weakness that diminishes the value of tests. Have a large number of diverse cases can mitigate this, however, dynamically generating the values significantly increases the tests effectiveness. This is why things like fuzzing are so powerful.

You *can* use a single hard-coded value, but if the pattern is wrong and the test case is based on the pattern, it won't help catch real-world issues.

---

I have written a function to generate a password with inclusion policy.